Implemented Status Bar

kosmas2004 opened the pull request:
I implemented the Status Bar at the bottom of the settings frame and it updates automatically every time you change the hints mode or the difficulty mode. I hope you are satisfied with the results and I'm looking forward for your response.

John423214 committed:
Implemented Status Bar

MonkeyxDev commented:
Hi, thanks again for your contribution! I found a few small issues and one optional improvement suggestion:

1. Status frame resizing issue

When the status text becomes longer, the frame automatically expands to fit the text.
Could you please update it so that the frame keeps a fixed size?

2. “updated to …” text stays and doesn’t reset

When switching settings multiple times, the “updated to …” message remains on the label.
Could you make it so the text resets or updates properly each time?
For example, clear or replace the previous status message before setting a new one.

3. Optional suggestion (UI/UX improvement)

Right now the status text appears inline at the bottom-left.
If you want, you could improve readability by changing this to a small titled section.
Something like a box with a title:

Details
---------
Hints: On
Difficulty: Easy
Theme: Dark
Tries : 20

Or whatever name you prefer (Details / Info / Status / Defaults).
This is optional, but it would make the UI neater and easier to read.

Note : could fix these issues myself, but since you said you want to improve and learn, I would like you to handle them so you can practice and grow. 🙂

kosmas2004 commented:
Hi, I will fix issue 1 & 3, and I want to fix issue number 2 as well. But I don't completely understand what you have in mind. Would you like for the "update to " part of the status bar to disappear after what action. After a certain amount of time or after a certain input the user gives like when he guesses for the first time after he changed the settings?

MonkeyxDev commented:
About issue #2, here is exactly what I had in mind:

When a setting is changed, the status bar shows the “updated to …” message, but this message should not stay permanently.
I want it to automatically disappear after 2–3 seconds.

The reason is that I want the user to feel that the UI is dynamic and responsive.
If the “updated to …” text stays there forever, it feels static and not very clean.
But if it disappears after a short moment, the user understands that the update happened and the interface continues to show the normal status information.
You can use after() method to remove the text after a delay.
like This : 
 show the update message
status_label.configure(text="Updated to ...")

 after 2000 ms (2 seconds), clear or restore normal text
status_label.after(2000, lambda: status_label.configure(text="Hints: On"))

kosmas2004 commented:
Perfect, thanks for the clarification. Most likely I will have it ready by the end of Tuesday. You will hear from me then. Have a nice day.

John423214 committed:
Fixed the 3 issues.

kosmas2004 commented:
Hi!

Issue 1: I fixed it using wraplength whenever the text  changes. This contains the length of the text up to a certain point. It distorts the text if you change it into something too big. But for this specific instance it works fine and in the future if you want to change the text into something bigger the wraplength value can be fine tuned to work for the specific text.

Issue 2: As you see i fixed this issue with the code you gave me, which was pretty useful, thanks.

Issue 3: I made an extra Label to use it as the title and modified the original Label to have a specific width and height to become a box.
